src/imap_mag/util/BinaryHelper.py

```diff
@@ -0,0 +1,83 @@
+import io
+import os
+from datetime import date
+from pathlib import Path
+
+import ccsdspy
+from ccsdspy.utils import iter_packet_bytes
+from rich.progress import Progress
+
+from imap_mag.util.TimeConversion import TimeConversion
+
+
+class BinaryHelper:
+    """
+    Class to handle binary data processing.
+    """
+
+    @staticmethod
+    def get_apids_and_days(binary_file: Path) -> dict[int, set[date]]:
```

Comment:
How about an OO version instead: `CCSDSBinaryPacketFile(file)` so you can reuse the packet definition?
and then method `get_apids_and_days()`